src/pykeen/nn/combination.py

```diff
@@ -0,0 +1,309 @@
+    parameters, $f_{gate}$ and $f_{hidden}$ are activation functions, defaulting to sigmoid and tanh, $\odot$ denotes
+    the element-wise multiplication, and $[x_e; x_l]$ the concatenation operation.
+
+    .. note ::
+
+        We can alternatively express the gate
+
+        .. math ::
+
+            z = f_{gate}(\mathbf{W}_e x_e + \mathbf{W}_l x_l + \mathbf{b})
+
+        as
+
+        .. math ::
+
+            z = f_{gate}(\mathbf{W}_{el} [x_e; x_l] + \mathbf{b})
+
+        with $\mathbf{W}_{el} \in \mathbb{R}^{(d_e + d_l) \times d_e}$.
+
+    Implementation based on https://github.com/SmartDataAnalytics/LiteralE/blob/master/model.py Gate class.
+    """
+
+    def __init__(
+        self,
+        entity_dim: int = 32,
+        literal_dim: Optional[int] = None,
+        input_dropout: float = 0.0,
+        gate_activation: HintOrType[nn.Module] = nn.Sigmoid,
+        gate_activation_kwargs: Optional[Mapping[str, Any]] = None,
+        hidden_activation: HintOrType[nn.Module] = nn.Tanh,
+        hidden_activation_kwargs: Optional[Mapping[str, Any]] = None,
+    ) -> None:
+        """Instantiate the module.
+
+        :param entity_dim:
+            the dimension of the entity representations.
+        :param literal_dim:
+            the dimension of the literals; defaults to entity_dim
+        :param input_dropout:
+            the dropout to use
+        :param gate_activation:
+            the activation to use on the gate, or a hint thereof
+        :param gate_activation_kwargs:
+            the keyword arguments to be used to instantiate the `gate_activation` if
+            a class or name is given instead of a pre-instantiated activation module
+        :param hidden_activation:
+            the activation to use in the hidden layer, or a hint thereof
+        :param hidden_activation_kwargs:
+            the keyword arguments to be used to instantiate the hidden activation if
+            a class or name is given instead of a pre-instantiated activation module
+        """
+        super().__init__()
+        literal_dim = literal_dim or entity_dim
+        self.dropout = nn.Dropout(input_dropout)
+        # the gate
+        self.gate = ConcatProjectionCombination(
+            input_dims=[entity_dim, literal_dim],
+            output_dim=entity_dim,
+            bias=True,
+            activation=gate_activation,
+            activation_kwargs=gate_activation_kwargs,
+        )
+        # the combination
+        self.combination = ConcatProjectionCombination(
+            input_dims=[entity_dim, literal_dim],
+            output_dim=entity_dim,
+            bias=True,
+            activation=hidden_activation,
+            activation_kwargs=hidden_activation_kwargs,
+        )
+
+    # docstr-coverage: inherited
+    def forward(self, xs: Sequence[torch.FloatTensor]) -> torch.FloatTensor:  # noqa: D102
+        assert len(xs) == 2
```

Comment:
is this still necessary?

Reply:
yes, because the initialization of gate / combination is hard-coded to be a ConcatProjectionCombination of two